test/integration/cli/server-process-cli-test.coffee

```diff
@@ -90,6 +90,26 @@ describe 'CLI - Server Process', ->
       it 'should exit with status 0', ->
         assert.equal dreddCommandInfo.exitStatus, 0
 
+    describe 'When it fails to start', ->
```

Comment:
I just realized there's one more side to the problem: The process starts, but Dredd isn't able to connect to it using HTTP. Looking into coverage, this scenario seems to be untested:

<img width="942" alt="image" src="https://user-images.githubusercontent.com/283441/29169360-fbd7923c-7dd3-11e7-91ee-9d9f8cf68aec.png">

_transaction-runner.coffee_

-----

I hope it would still work as expected (Dredd fails all tests and it ends with exit code 1), but it would be awesome if we had a test for that as well. What do you think?

It should be fairly easy to test, with the same approach, just spawning an existing script, which hangs the same way as server would do (endless `setTimeout` loop?), but which doesn't handle HTTP in any way. Maybe there's even already a script like that in `test/fixtures`, I haven't take a look.